interceptors/realip/realip.go

```diff
@@ -108,25 +130,56 @@ func (s *serverStream) Context() context.Context {
 	return s.ctx
 }
 
+// Opts specifies the options of the middleware
+type Opts struct {
```

Comment:
This is great, but we generally use functional options to configure the interceptors. Do you mind rewriting this in the functional option style? Thanks!

Reply:
Oh indeed, I had to look at this before. Now this is reworked to functional options.